src/macros.rs

```diff
@@ -267,3 +267,46 @@ macro_rules! define_casts {
         )+
     }
 }
+
+#[macro_export]
+macro_rules! cfg_feature_enabled {
+    ($name:tt) => (
+        {
+            #[cfg(target_feature = $name)]
+            {
+                true
+            }
+            #[cfg(not(target_feature = $name))]
+            {
+                __unstable_detect_feature!($name)
+            }
+        }
+    )
+}
+
+/// On ARM features are only detected at compile-time using
+/// cfg(target_feature), so if this macro is executed the
+/// feature is not supported.
+#[cfg(any(target_arch = "arm",
+          target_arch = "aarch64"))]
+#[macro_export]
+macro_rules! __unstable_detect_feature {
+    ($t:tt) => { false }
```

Comment:
Could this and the fallback implementation (and x86) do something like:

```rust
macro_rules! __unstable_detect_feature {
    // ...
    ($t:tt) => (compile_error!(concat!("unknown target feature: ", $t)));
}
```

Reply:
`compile_error!` !!! I did not knew this existed! Sure!